### PR TITLE
Allow configuration of hostname

### PR DESCRIPTION
**Pull request by SamSaffron**
These changes have been running in production for a while now, allow hostname configuration on containers.

### CODE REVIEW

File: lib/moby_derp/pod.rb [hunk 1]
```diff
@@ -38,9 +38,9 @@ def run
 				begin
 					MobyDerp::Container.new(pod: self, container_config: cfg).run
 				rescue MobyDerp::ContainerError => ex
-					raise MobyDerp::ContainerError,
-					      "error while running container #{cfg.name}: #{ex.message}",
-							ex.backtrace
+					@logger.error(logloc) {
```
Comment (mpalmer) [inline lib/moby_derp/pod.rb:41]: This change seems unrelated to the purpose of the other changes in this PR; it would be better to submit as a separate PR as I have some potential reservations, and I don't want to hold up these other changes debating it.

File: spec/moby_derp/system_config_spec.rb [hunk 2]
```diff
@@ -97,8 +97,6 @@
 			"mount_root:\n- one\n- two",
 		"when the mount_root specified in the config file isn't an absolute path" =>
 			"mount_root: relative/path",
-		"when the mount_root specified in the config file doesn't exist" =>
```
Comment (mpalmer) [inline spec/moby_derp/system_config_spec.rb:100]: This change seems unrelated to the general purpose of the PR.

File: lib/moby_derp/system_config.rb [hunk 3]
```diff
@@ -17,6 +17,7 @@ def initialize(config_data_or_filename, moby_info, logger)
 				raise ArgumentError, "Unsupported type for config_data_or_filename parameter"
 			end
 
+			@host_hostname        = @config["host_hostname"] || Socket.gethostname
```
Comment (mpalmer) [inline lib/moby_derp/system_config.rb:20]: Use of the `Socket` constant requires `require "socket"`.  There also doesn't appear to be a new test verifying that the default value of this system configuration parameter comes from `Socket.gethostname`.

File: README.md [hunk 4]
```diff
@@ -124,6 +124,9 @@ The keys are:
   "localhost"; so pointing to your local caching resolver using `127.0.0.1`
   will not end in happiness and puppies.
 
+* **`host_hostname`**: the hostname to use when generating the container
+  hostname.
```
Comment (mpalmer) [inline README.md:128]: It would be helpful to explain the circumstances under which this might be useful.

**Review by mpalmer — CHANGES_REQUESTED** [review]

**Comment by SamSaffron** [comment]
No worries, let me close this one, I will get Daniel to open stuff in a more orderly fashion, I don't want there to be too much (or any)  drift between upstream and the Discourse fork.
